staging/src/k8s.io/apiserver/pkg/apis/apiserverinternal/validation/validation.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/apiserverinternal"
+)
+
+// ValidateStorageVersion validate the storage version object.
+func ValidateStorageVersion(sv *apiserverinternal.StorageVersion) field.ErrorList {
```

Comment:
need validation on the conditions.  hopefully there's something stock

Reply:
Added. 
[validatePodConditions](https://github.com/kubernetes/kubernetes/blob/f372c5417b2053827e14bc4086ee7828ba674d53/pkg/apis/core/validation/validation.go#L3877) is the only example in the code base.